Dockerfile

```diff
@@ -25,6 +30,316 @@ RUN pacman -Syu \
   texlive-bin \
   wget \
   --noconfirm
+  
+###############################################################################
+# AUR DEPENDENCIES
+###############################################################################
+
+# we need opam, perl-critic, verilator
+
+# package (sub) dependencies
+RUN pacman -Syu \
+  base-devel \
+  bison \
+  boost \
+  cmake \
+  ocaml \
+  perl-class-inspector \
+  perl-clone \
+  perl-config-tiny \
+  perl-email-address \
+  perl-exception-class \
+  perl-file-remove \
+  perl-file-sharedir \
+  perl-file-which \
+  perl-io-string \
+  perl-list-moreutils \
+  perl-module-build \
+  perl-module-pluggable \
+  perl-module-runtime \
+  perl-params-util \
+  perl-readonly \
+  perl-role-tiny \
+  perl-sub-exporter \
+  perl-test-deep \
+  perl-test-fatal \
+  perl-test-pod \
+  perl-test-pod-coverage \
+  perl-tidy \
+  re2c \
+  scons \
+  --noconfirm
+  
+# create build user and group 
+RUN groupadd -r build
+RUN useradd -r -g build build
+RUN mkdir /tmp/build
+
+# ensure permissions
+RUN chown -R build:build /tmp/build && \
+    chmod -R 771 /tmp/build
+
+# opam and its dependencies, DO NOT CHANGE ORDER! 
+#************************************************
+
+# build opam dependency (clasp)
+USER build 
+WORKDIR /tmp/build
+RUN git clone https://aur.archlinux.org/clasp.git
+WORKDIR /tmp/build/clasp
+RUN makepkg --noconfirm
+USER root 
+WORKDIR /tmp/build/clasp
+RUN pacman --noconfirm -U *.tar.xz
+
+# build opam dependency (gringo)
+USER build 
+WORKDIR /tmp/build
+RUN git clone https://aur.archlinux.org/gringo.git
+WORKDIR /tmp/build/gringo
+RUN makepkg --noconfirm
+USER root 
+WORKDIR /tmp/build/gringo
+RUN pacman --noconfirm -U *.tar.xz
+
+# build opam dependency (aspcud)
+USER build 
+WORKDIR /tmp/build
+RUN git clone https://aur.archlinux.org/aspcud.git
+WORKDIR /tmp/build/aspcud
+RUN makepkg --noconfirm
+USER root 
+WORKDIR /tmp/build/aspcud
+RUN pacman --noconfirm -U *.tar.xz
+
+# build (opam)
+USER build 
+WORKDIR /tmp/build
+RUN git clone https://aur.archlinux.org/opam.git
+WORKDIR /tmp/build/opam
+RUN makepkg --noconfirm
+USER root 
+WORKDIR /tmp/build/opam
+RUN pacman --noconfirm -U *.tar.xz
```

Comment:
```###############################################################################
# SETTING UP YAOURT
###############################################################################

# synchronizing package list 
RUN pacman --noconfirm -Sy

# install base-devel and yajl 
RUN pacman --noconfirm -S base-devel
RUN pacman --noconfirm -S yajl

# create yaourt user and group RUN groupadd -r yaourt
RUN useradd -r -g yaourt yaourt
RUN mkdir /tmp/yaourt

RUN chown -R yaourt:yaourt /tmp/yaourt && \
    chmod -R 771 /tmp/yaourt
USER yaourt 
WORKDIR /tmp/yaourt

# cloning and install yaourt dependency (package-query) 
RUN git clone https://aur.archlinux.org/package-query.git 
WORKDIR /tmp/yaourt/package-query
RUN makepkg --noconfirm
USER root 
WORKDIR /tmp/yaourt/package-query
RUN pacman --noconfirm -U *.tar.xz

# cloning and install yaourt 
USER yaourt 
WORKDIR /tmp/yaourt
RUN git clone https://aur.archlinux.org/yaourt.git
WORKDIR /tmp/yaourt/yaourt
RUN makepkg --noconfirm
USER root 
WORKDIR /tmp/yaourt/yaourt
RUN pacman --noconfirm -U *.tar.xz

# remove user yaourt (for security reasons) 
RUN userdel -r yaourt

RUN mkdir -p /tmp/Package/ && chown yaourt /tmp/Package
USER yaourt
RUN cd /tmp/Package && pwd && ls -al && yaourt --getpkgbuild aur/PACKAGE && cd taskd && makepkg --pkg PACKAGE 
USER root
RUN pacman -U --noconfirm /tmp/Package/PACKAGE/PACKAGE-1.1.1-1-x86_64.pkg.tar.xz```

yeah this saves you 2 lines per package
not even yaourt is allowed to build stuff as administrator -.-